packages/api/src/router/github/create-github-org.ts

```diff
@@ -0,0 +1,225 @@
+            .flat(),
+        ),
+        inArray(
+          workspace.slug,
+          parsedConfigFiles
+            .map((d) => d.content.deployments.map((d) => d.workspace))
+            .flat(),
+        ),
+      ),
+    );
+
+  const insertedConfigFiles = await db
+    .insert(githubConfigFile)
+    .values(
+      parsedConfigFiles.map((d) => ({
+        ...d,
+        workspaceId: org.workspaceId,
+        organizationId: org.id,
+        repositoryName: d.repository.name,
+      })),
+    )
+    .returning();
+
+  const deployments = parsedConfigFiles
+    .map((cf) =>
+      cf.content.deployments.map((d) => {
+        const info = deploymentInfo.find(
+          (i) => i.system.slug === d.system && i.workspace.slug === d.workspace,
+        );
+        if (info == null) throw new Error("Deployment info not found");
+        const { system, workspace } = info;
+
+        return {
+          ...d,
+          systemId: system.id,
+          workspaceId: workspace.id,
+          description: d.description ?? "",
+          githubConfigFileId: insertedConfigFiles.find(
+            (icf) =>
+              icf.path === cf.path && icf.repositoryName === cf.repository.name,
+          )?.id,
+        };
+      }),
+    )
+    .flat();
+
+  await db
+    .insert(deployment)
+    .values(deployments)
+    .onConflictDoUpdate({
+      target: [deployment.systemId, deployment.slug],
+      set: {
+        githubConfigFileId: sql`excluded.github_config_file_id`,
+      },
+    });
+};
+
+export const createNewGithubOrganization = async (
+  db: Tx,
+  githubOrganizationConfig: GithubOrganizationInsert,
+) =>
+  db.transaction(async (db) => {
+    const org = await db
+      .insert(githubOrganization)
+      .values(githubOrganizationConfig)
+      .returning()
+      .then(takeFirst);
+
+    const installation = await octokit?.apps.getInstallation({
+      installation_id: org.installationId,
+    });
+    if (installation == null) throw new Error("Failed to get installation");
+
+    const installationOctokit = getOctokitInstallation(installation.data.id);
+    const installationToken = (await installationOctokit.auth({
+      type: "installation",
+      installationId: installation.data.id,
+    })) as { token: string };
+
+    const configFiles = await Promise.all([
+      installationOctokit.search.code({
+        q: `org:${org.organizationName} filename:ctrlplane.yaml`,
+        per_page: 100,
+        headers: {
+          "X-GitHub-Api-Version": "2022-11-28",
+          authorization: `Bearer ${installationToken.token}`,
+        },
+      }),
+      installationOctokit.search.code({
+        q: `org:${org.organizationName} filename:ctrlplane.yml`,
+        per_page: 100,
+        headers: {
+          "X-GitHub-Api-Version": "2022-11-28",
+          authorization: `Bearer ${installationToken.token}`,
+        },
+      }),
+    ]).then((responses) => {
+      return [...responses[0].data.items, ...responses[1].data.items];
+    });
```

Comment:
1 line, no need for return

be more descriptive, give these vars names for the data they are providing

```suggestion
    ]).then(([valueA, valueB]) => 
      [...valueA.data.items, ...valueBdata.items]
    );
```